designs/conditional-controllers.md

```diff
@@ -0,0 +1,174 @@
+Informer Removal / Controller Lifecycle Management
+==========================
+
+## Summary
+
+Enable fine-grained control over the lifecycle of a controller, including the
+ability to start/stop/restart controllers and their caches by exposing a way to
+remove individual informers from the cache and working around restrictions that
+currently prevent controllers from starting multiple times.
+
+## Background/Motivation
+
+Currently, the user does not have much controller over the lifecycle of a
+controller. The user can add controllers to the manager and add informers to the
+cache, but there is no way to remove either of these.
+
+Additionally there are restrictions that prevent a controller from running
+multiple times such as the clearing the watches slice for a controller after it
+has started. The effect of this is that users have no clean way of restarting controllers
+after they have stopped them.
+
+Greater detail is given in the "Future Work / Use Cases" section, but a summary
+of the motivating use-cases for the proposal is anytime a controller's watched resource
+can be installed/uninstalled unexpectedly or whenever the administrator of a
+controller is different from the administrator of the cluster (and thus has no
+control over which resources are installed).
+
+## Goals
+
+An implementation of the minimally viable hooks needed in controller-runtime to
+enable controller adminstrators to start, stop, and restart controllers and their caches.
```

Comment:
non-goals says that you don't want to add support to start/stop on arbitrary conditions. Please explicitly mention the conditions this is supposed to support

Reply:
Sorry, I don't think the non-goals are clear, basically I was just trying to say that the goals are just to provide a hook, non-goals are an actual implementation of the hooks.